cmake/CMakeLists.txt

```diff
@@ -46,6 +46,11 @@ else()
 set(CMAKE_CXX_STANDARD 17)
 endif()
 
+if (MSVC)
+  #  Make sure Visual Studio sets __cplusplus macro correctly: https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus")
```

Comment:
Could you please explain why this change is needed for this PR? Is it only required by cutlass 3.5.0?

Reply:
See https://github.com/NVIDIA/cutlass/issues/1403.
Right now it only impacts cutlass build using nvcc from CUDA >= 12.4.

Reply:
Just FYI, https://github.com/microsoft/onnxruntime/issues/20923
We are working on a fix for TRT EP in response to adding `/Zc:__cplusplus`